evalplus/codegen.py

```diff
@@ -152,6 +152,45 @@ def run_codegen(
         "perf-CoT",
     ]
 
+    # Make dir for codes generated by each model
+    identifier = model.strip("./").replace("/", "--") + f"_{backend}_temp_{temperature}"
+    if evalperf_type:
+        identifier += f"-{evalperf_type}"
+
+    target_path = os.path.join(root, dataset, identifier)
+    if jsonl_fmt:
+        target_path += ".jsonl"
+    else:
+        os.makedirs(target_path, exist_ok=True)
+
+    all_tasks_complete = False
+    if jsonl_fmt and os.path.isfile(target_path):
+        task_counts = {}
+        with open(target_path, 'r') as f:
+            for line in f:
+                if not line.strip():
+                    continue
+                data = json.loads(line)
+                task_id = data['task_id']
+                task_counts[task_id] = task_counts.get(task_id, 0) + 1
+
+            if dataset == "humaneval":
+                current_dataset = get_human_eval_plus(version=version)
+            elif dataset == "mbpp":
+                current_dataset = get_mbpp_plus(version=version)
+            elif dataset == "evalperf":
+                original_dataset = {**get_human_eval_plus(), **get_mbpp_plus()}
+                current_dataset = {k: original_dataset[k] for k in get_evalperf_data()}
+                assert id_range is None, "id_range not supported for evalperf"
+            else:
+                raise ValueError(f"Invalid dataset {dataset}")

```

Comment:
Would be nice to unify this with L40-L50. We just pass a dataset to the `codegen` function